packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -132,6 +136,92 @@ export class SamplerOperations {
             .catch(/* do nothing */);
     }
 
+    public async getTokenDecimalsAsync(tokens: Address[]): Promise<number[]> {
+        return (await this.rpcSamplerClient.getTokensAsync(tokens)).map(t => t.decimals);
+    }
+
+    public async getSellQuotesAsync(
+        sources: ERC20BridgeSource[],
+        makerToken: string,
+        takerToken: string,
+        takerAmount: BigNumber,
+    ): Promise<Array<Array<DexSample<FillData>>>> {
+        const rpcLiquidityRequests: RpcLiquidityRequest[] = sources.map(source => {
+            return {
+                tokenPath: [makerToken, takerToken],
+                inputAmount: takerAmount.toString(),
+                source,
+                demand: true,
+            };
+        });
+        const rpcLiquidityResponse = await this.rpcSamplerClient.getSellLiquidityAsync(rpcLiquidityRequests);
+        const dexQuotes: Array<Array<DexSample<FillData>>> = rpcLiquidityResponse.map((response, i) => {
+            const dexSample: Array<DexSample<FillData>> = response.liquidityCurve.map((point, j) => {
+                const fillData: DexSample = {
+                    source: response.source,
+                    fillData: point.encodedFillData,
```

Comment:
So now these will be
```suggestion
                    encodedFillData: point.encodedFillData,
```
And we can get rid of the code that encodes them.
We can comment out multihop and all VIPs for now since they actually inspect `fillData` at the moment.